manticore/ethereum/account.py

```diff
@@ -48,93 +48,90 @@ def __str__(self):
 
 
 class EVMContract(EVMAccount):
-    """ An EVM account
-
-        Note: The private methods of this class begin with a double underscore to
-        avoid function name collisions with Solidity functions that begin with
-        a single underscore
     """
+    An EVM account
 
+    Note: The private methods of this class begin with a double underscore to avoid function
+    name collisions with Solidity functions that begin with a single underscore.
+    """
     def __init__(self, default_caller=None, **kwargs):
-        """ Encapsulates a contract account.
-            :param default_caller: the default caller address for any transaction
+        """
+        Encapsulates a contract account.
 
+        :param default_caller: the default caller address for any transaction
         """
         super().__init__(**kwargs)
         self._default_caller = default_caller
         self._hashes = None
+        self._initialized = False
 
     def add_function(self, signature):
         func_id = ABI.function_selector(signature)
         func_name = str(signature.split('(')[0])
-        if func_name.startswith('__') or func_name in {'add_function', 'address', 'name_'}:
-            # TODO(mark): is this actually true? is there anything actually wrong with a solidity name beginning w/ an underscore?
-            raise EthereumError("Function name ({}) is internally reserved".format(func_name))
+        if func_name.startswith('_EVMContract__') or func_name in {'add_function', 'address', 'name_'}:
```

Comment:
Hmm this doesn't protect against making a function named: `_default_caller`, `_hashes`, `_initialized`.

I think I will make those protected (use `__` instead of `_`) so they are not accessible from the outside at all (and so their real attribute name starts with `_EVMContract__`).

Reply:
![image](https://user-images.githubusercontent.com/10009354/50119164-2bfbb200-0252-11e9-9506-ea602fb66042.png)
Yes! This + checking for attributes that matches the function names.